Update configs to allow for return to main/exit and repeat menu

jrrose opened the pull request:
Adds configuration information necessary for adding the following DTMF buttons:

'*' - repeats the last menu.
'#' - exits from the change folders prompt. If already in the main menu, plays bye and hangs up.

jrrose committed:
Update configs to allow for return to main/exit and repeat menu

jrrose committed:
Review feedback for repeatMenu and previousMenu

Pull Request: https://github.com/asterisk/node-voicemail-main-fsm/pull/2
Pull Request: https://github.com/asterisk/node-voicemail-main/pull/1
Pull Request: https://github.com/asterisk/node-voicemail-mailbox/pull/1

samuelg commented on config.json:
```diff
@@ -219,8 +224,10 @@
-        "action": "submit"
+        "regex": "\\d",
+        "action": "submit",
+        "#": "previousMenu",
```
The code that handles converting DTMF input to FSM actions expects either a key/value pair where the key is the DTMF and the value is the action to take or a structure with a regex and an action, not sure how it will handle both at the same time.

samuelg replied:
After reviewing the code in asterisk/node-voicemail-main-fsm, it looks like that will actually work as intended but I still think it's confusing from the point of view of the config.

I think we should think about this some more to see if there's a better way to define the config that is less confusing and works for all cases (I'm sure this would involve slightly modifying the asterisk/node-voicemail-main-fsm dtmf handler code).

Commit:
Changed config format for matching dtmf with regex.

Changed the configuration format for matching dtmfs to an action using
a regular expression so it can be mixed with simple dtmf/action mappings.

jrrose committed:
Merge pull request #2 from samuelg/dtmfconfig

Update to match new regex format for dtmfs.
